vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/GraphQLSocketHandler.java

```diff
@@ -0,0 +1,16 @@
+package io.vertx.ext.web.handler.graphql;
+
+import graphql.GraphQL;
+import io.vertx.core.Handler;
+import io.vertx.core.http.ServerWebSocket;
+import io.vertx.ext.web.handler.graphql.impl.GraphQLSocketHandlerImpl;
+
+public interface GraphQLSocketHandler extends Handler<ServerWebSocket> {
+
+  GraphQLSocketHandler endHandler(Handler<ServerWebSocket> endHandler);
```

Comment:
What usages do you have in mind for this callback?

A callback for handshake would be useful since, for example, users could extract an auth token from headers

Reply:
The callback is called when a websocket connection is closed. Maybe the Handler can evolve and callbacks can be added when they are asked for. So, I can delete the endHandler.

Due to the handshake is an HTTP call, I think manipulating the HTTP request should be done adding handlers to the path in a previous order. For regular endpoints it is done in that way and having a unique way to do things is a good practice.

Reply:
>The callback is called when a websocket connection is closed. Maybe the Handler can evolve and callbacks can be added when they are asked for. So, I can delete the endHandler.

Ok, let's start without it.

>Due to the handshake is an HTTP call, I think manipulating the HTTP request should be done adding handlers to the path in a previous order. For regular endpoints it is done in that way and having a unique way to do things is a good practice.

Fair enough. We can revisit later if it proves to be inconvenient. Make sure to document how to authenticate a pure WS client though.